lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -122,18 +122,23 @@ def controller_get_shared_from_options options
         end
 
         def create_disk disk, index = 0, operation = :add, controller_key = 1000
+          if (index > 6) then
+            _index = index + 1
+          else
+            _index = index
+          end
```

Comment:
Isn´t this the case to only increment if index equals 7?

Maybe something like:

```ruby
index += 1 if index == 7
```

would fit better. Make sense?